app/fn-guru/src/main/frontend/src/pages/app/exec-list/components/hook.ts

```diff
@@ -0,0 +1,41 @@
+import {ExecList, ExecListItem} from "@/types";
+import {useAuth} from "@/hook/auth.ts";
+import {useExecList, useTriggerList} from "@/hook";
+import {useCallback, useEffect, useState} from "react";
+
+
+type ExecsWithTriggersAction = (namespaceId: string, abortController?: AbortController) => void
+export const useExecsWithTriggers = (): [ExecsWithTriggersAction, ExecList, boolean, Error] => {
+    const [auth] = useAuth()
+    const [listTriggers, triggerList, , triggerError] = useTriggerList()
+    const [listExecs, execList, , execError] = useExecList()
+    const [loading, setLoading] = useState(true)
+    const [execsWithTriggerStatus, setExecsWithTriggerStatus] = useState<ExecList>(null)
+
+    const fn = useCallback<ExecsWithTriggersAction>(
+        async (namespaceId, abortController ?) => {
+            listTriggers(namespaceId, abortController)
+            listExecs(namespaceId, abortController)
+        }, [auth]
+    )
+
+    useEffect(() => {
+        if (triggerList && execList) {
+            const copy = {...execList};
+            copy.execs.forEach(exec => {
+                if ((exec.invocation.class === "Schedule" || "Event" || "Hook" || "Endpoint") && exec.func != null) {
+                    const execTrigger = triggerList.triggers.find(trigger => trigger.func.id === exec.func.id);
+                    exec.trigger = {
+                        id: execTrigger.id,
+                        status: execTrigger.status
+                    }
+                }
+            });
+            setExecsWithTriggerStatus(copy)
+        }
+        setLoading(false)
+    }, [triggerList, execList]);
```

Comment:
I don't think that's the implementation we need for this - we should add this to the backend which trigger triggered an execution.

Benefits of doing so - we save 1 http call - there can be multiple trigger for the same function --> your implementation does not take care of this

Reply:
We already have the invocation class - we just need to add the trigger id to the invocation - the moment we rendering it to ApiExec... we fetch the current trigger status and return it in the exec model